internal/controllers/user_devices_controller.go

```diff
@@ -231,75 +231,80 @@ func (udc *UserDevicesController) dbDevicesToDisplay(ctx context.Context, device
 		var nft *VehicleNFTData
 		var credential *VINCredentialData
 		pu := []PrivilegeUser{}
-		if !d.TokenID.IsZero() {
-			nftStatus := d.R.MintRequest
-			nft = &VehicleNFTData{
-				Status: nftStatus.Status,
-			}
-			if nftStatus.Hash.Valid {
-				hash := hexutil.Encode(nftStatus.Hash.Bytes)
-				nft.TxHash = &hash
-			}
 
-			nft.TokenID = d.TokenID.Int(nil)
-			nft.TokenURI = fmt.Sprintf("%s/v1/vehicle/%s", udc.Settings.DeploymentBaseURL, nft.TokenID)
-			if d.OwnerAddress.Valid {
+		if !d.TokenID.IsZero() || d.R.MintRequest != nil {
```

Comment:
Still display this section if there's no token id. Need to tell the frontend, in particular, how things are going during minting.